### PR TITLE
completes step 6

### PR DESCRIPTION
**Pull request by jcainuk**
completes step 6

### CODE REVIEW

File: app.test.js [hunk 1]
```diff
@@ -127,3 +126,50 @@ describe("/api/articles/:article_id", () => {
...
+          expect(Object.keys(response.body.comments[0])).toEqual(
+            expect.arrayContaining([
+              "comment_id",
+              "votes",
+              "created_at",
+              "author",
+              "body",
+              "article_id"
+            ])
+          );
```
Comment (Slyose) [inline app.test.js]: I would say just checking the first result has the keys you expect doesn't really prove that all of the articles have the keys you expect also. We also aren't checking what the values are or what data type they should be.

Consider trying to use a forEach here to look at each article and then using something like hasOwnProperty 😄

Something along the lines of
articles.forEach((article) => {
expect(article).hasOwnProperty("article_id", expect.any(Number))

etc. etc. for each property you expect to find 😄

File: app.test.js [hunk 2]
```diff
@@ -127,3 +126,50 @@ describe("/api/articles/:article_id", () => {
...
+          });
+        });
+    });
+    test("200: article comments should be served in an array with the most recent comments first", () => {
+      return request(app)
+        .get("/api/articles/1/comments")
+        .expect(200)
+        .then((response) => {
+          const { comments } = response.body;
+          expect(Array.isArray(comments)).toBe(true);
```
Comment (Slyose) [inline app.test.js]: A .length check would be better than an .isArray here. An empty array would pass your toBeSortedByTest, as there's nothing to compare. Checking the length will ensure that an empty array won't pass!

File: app.test.js [hunk 3]
```diff
@@ -127,3 +126,50 @@ describe("/api/articles/:article_id", () => {
...
+
+    test("GET:404 sends an appropriate and error message when given a valid but non-existent id", () => {
+      return request(app)
+        .get("/api/articles/9000/comments")
+        .expect(404)
+        .then((response) => {
+          expect(response.body.msg).toBe("article does not exist");
+        });
+    });
+  });
```
Comment (Slyose) [inline app.test.js:179]: Also worth having a bad request test here too!
What happens when someone requests  `.get("/api/articles/banana/comments")` ?

**Review by Slyose — COMMENTED** [review]
A few tweaks to consider before merging this, but you're close!

**Comment by jcainuk** [comment]
Thank you for your feedback, I have now pushed the required changes :)

**Comment by Harrison365** [comment]
Great changes 🎉 
Merge away ✅

**Pull request merged**